app/src/main/java/Storages/Implementations/IngredientStorageImpl.java

```diff
@@ -69,8 +69,17 @@ public String type() {
 			}
 
 		}
-
 		return found;
 	}
 
+	@Override
+	public String ingredientString() {
+		StringBuilder stringVal = new StringBuilder();
+		for (Ingredient ingredient : this.ingredients()) {
+			stringVal.append(ingredient.name());
+			stringVal.append(",");
+		}
+		stringVal.delete(stringVal.length(), stringVal.length());
+		return new String(stringVal);
+	}
```

Comment:
```
this.ingredients().stream().map(Ingredient::name).collect(Collectors.toList());
```
Will return a list of the names of ingredients.

```
String.join(",", list)
```
Will join all strings in a list with a comma.